src/app/components/reports/pages/event-summary/new/new.component.ts

```diff
@@ -0,0 +1,250 @@
+import { Site } from "@models/Site";
+import { Observable } from "rxjs";
+import { InnerFilter } from "@baw-api/baw-api.service";
+import { IPageInfo } from "@helpers/page/pageInfo";
+import {
+  ShallowSitesService,
+  siteResolvers,
+} from "@baw-api/site/sites.service";
+import {
+  regionResolvers,
+  ShallowRegionsService,
+} from "@baw-api/region/regions.service";
+import {
+  retrieveResolvers,
+  hasResolvedSuccessfully,
+} from "@baw-api/resolver-common";
+import { Region } from "@models/Region";
+import { Project } from "@models/Project";
+import { AudioEventProvenance } from "@models/AudioEventProvenance";
+import {
+  reportMenuItems,
+  reportCategories,
+} from "@components/reports/reports.menu";
+import { Tag } from "@models/Tag";
+import { TagsService } from "@baw-api/tag/tags.service";
+import { AudioEventProvenanceService } from "@baw-api/AudioEventProvenance/AudioEventProvenance.service";
+import { StrongRoute } from "@interfaces/strongRoute";
+import {
+  excludePropertyValues,
+  filterAnd,
+  filterModel,
+  propertyFilter,
+} from "@helpers/filters/filters";
+import { DateTimeFilterModel } from "@shared/date-time-filter/date-time-filter.component";
+import { DateTime } from "luxon";
+import { Id } from "@interfaces/apiInterfaces";
+import { AbstractModel } from "@models/AbstractModel";
+import {
+  BucketSize,
+  EventSummaryReportParameters,
+} from "../EventSummaryReportParameters";
+
+const projectKey = "project";
+const regionKey = "region";
+const siteKey = "site";
+
+@Component({
+  selector: "baw-new-summary-report",
+  templateUrl: "./new.component.html",
+})
+class NewEventReportComponent extends PageComponent implements OnInit {
+  public constructor(
+    protected sitesApi: ShallowSitesService,
+    protected regionsApi: ShallowRegionsService,
+    protected provenanceApi: AudioEventProvenanceService,
+    protected tagsApi: TagsService,
+    private route: ActivatedRoute
+  ) {
+    super();
+    this.model = new EventSummaryReportParameters();
+  }
+
+  public project: Project;
+  public region?: Region;
+  public site?: Site;
+  public model = new EventSummaryReportParameters();
+
+  protected get componentTitle(): string {
+    if (this.site) {
+      return this.site.isPoint
+        ? `Point: ${this.site.name}`
+        : `Site: ${this.site.name}`;
+    } else if (this.region) {
+      return `Site: ${this.region.name}`;
+    }
+
+    // it can be assumed that all reports will be generated from at least at the project level
+    // therefore, if no sites or regions are specified, we can assume that the report is being generated at the project level
+    return `Project: ${this.project.name}`;
+  }
+
+  protected get availableBucketSizes() {
+    return Object.keys(BucketSize);
+  }
+
+  public ngOnInit(): void {
+    const models = retrieveResolvers(this.route.snapshot.data as IPageInfo);
+
+    if (!hasResolvedSuccessfully(models)) {
+      return;
+    }
```

Comment:
There's no recovery for this case

I was under the assumption that this was a convention in our client so we'd soft fail in the case: the user loads a page -> internet drops out -> user loads a new page

However, looking at other examples of our code base, we don't use this convention.

I'll be removing this as:
a. It's not our convention
b. Hard failing is probably better

Reply:
You might be kinda right; I think our main view loads the error component when it detects a resolver failure. If that is our model though, the components themselves shouldn't have this if failed do nothing check.... it should be abstracted into the view logic for the main component. Otherwise not very DRY